astropy/units/quantity.py

```diff
@@ -461,12 +462,26 @@ def __array_prepare__(self, obj, context=None):
             # ensure we remember the scales we need
             result._converters = converters
 
+            if function in _UFUNCS_FILTER_WARNINGS:
+                # Filter out RuntimeWarning's caused by the ufunc being called on
+                # the unscaled quantity first (e.g., np.arcsin(15*u.pc/u.kpc))
+                self._catch_warnings = warnings.catch_warnings()
+                self._catch_warnings.__enter__()
+                warnings.filterwarnings('ignore',
+                                        message='invalid value encountered in',
+                                        category=RuntimeWarning)
+
         # unit output will get (setting _unit could prematurely change input
         # if obj is self, which happens for in-place operations; see above)
         result._result_unit = result_unit
+
         return result
 
     def __array_wrap__(self, obj, context=None):
+        if hasattr(self, '_catch_warnings'):
```

Comment:
Should have seen this before: just so the no-conversion case doesn't get slowed down, can you move this to l.504 (i.e., within the `if hasattr(obj, '_converters')` clause)...